MIT Press Books.js

```diff
@@ -0,0 +1,298 @@
+				contributors.forEach(function (contributorName) {
+					item.creators.push(ZU.cleanAuthor(contributorName, contributorType[1]));
+				});
+			}
+		});
+	});
+	
+	let series = ZU.xpathText(doc, '//p[@class = "book__series"][1]/a');
+	if (series) {
+		item.series = series.trim();
+	}
+	
+	let openAccessUrl = ZU.xpathText(doc, '//div[contains(@class, "open-access")]/a/@href');
+	if (openAccessUrl) {
+		if (openAccessUrl.endsWith('.pdf') || openAccessUrl.endsWith('.pdf?dl=1')) {
+			item.attachments.push({
+				url: openAccessUrl,
+				title: 'Full Text PDF',
+				mimeType: 'application/pdf'
+			});
+		}
+		else {
+			item.attachments.push({
+				url: openAccessUrl,
+				title: 'Open Access',
+				mimeType: 'text/html'
+			});
+		}
+	}
+	
+	item.complete();
+}
+
+function getSearchResults(doc, checkOnly) {
+	let rows = ZU.xpath(doc, '//ul[contains(@class, "results__list")]//a[@property = "name"]');
+	
+	if (checkOnly) {
+		return rows.length > 0;
+	}
+	
+	let items = {};
+	for (let i = 0; i < rows.length; i++) {
+		items[rows[i].href] = rows[i].text.trim();
+	}
+	return items;
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://mitpress.mit.edu/books/series/adaptive-computation-and-machine-learning-series",
+		"items": "multiple"
+	},
+	{
+		"type": "web",
+		"url": "https://mitpress.mit.edu/books/elements-causal-inference",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "Elements of Causal Inference: Foundations and Learning Algorithms",
+				"creators": [
+					{
+						"firstName": "Jonas",
+						"lastName": "Peters",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "Dominik",
+						"lastName": "Janzing",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "Bernhard",
+						"lastName": "Schölkopf",
+						"creatorType": "author"
+					}
+				],
+				"date": "2017-11-29",
+				"ISBN": "9780262037310",
+				"language": "en",
+				"libraryCatalog": "MIT Press Books",
+				"numPages": "288",
+				"place": "Cambridge, MA, USA",
+				"publisher": "MIT Press",
+				"series": "Adaptive Computation and Machine Learning series",
+				"shortTitle": "Elements of Causal Inference",
+				"attachments": [
+					{
+						"title": "Full Text PDF",
+						"mimeType": "application/pdf"
+					}
+				],
+				"tags": [],
+				"notes": [],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "https://mitpress.mit.edu/search?keywords=deep+learning",
+		"items": "multiple"
+	},
+	{
+		"type": "web",
+		"url": "https://mitpress.mit.edu/contributors/ian-goodfellow",
+		"items": "multiple"
+	},
+	{
+		"type": "web",
+		"url": "https://mitpress.mit.edu/books/sciences-artificial-reissue-third-edition-new-introduction-john-laird",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "The Sciences of the Artificial, Reissue Of The Third Edition With A New Introduction By John Laird",
```

Comment:
It looks like the edition (as well as volume) info is saved as part of the title. This would be good to try to split it of, if we can. This example here looks extremely challenging and therefore I am not sure that we should tackle it at all. But easier example are maybe
- Foundations of Machine Learning, Second Edition: https://mitpress.mit.edu/books/foundations-machine-learning-second-edition
- Collapse, Volume 8: https://mitpress.mit.edu/books/collapse-volume-8
- Introduction to Algorithms, Third Edition: https://mitpress.mit.edu/books/introduction-algorithms-third-edition

Can we fix at least these ans similar examples?